script.js

```diff
@@ -1,4 +1,381 @@
+  var suits = ["Spade", "Heart", "Club", "Diamond"];
+  var deck = [];
+
+  for (var i = 0; i < suits.length; i += 1) {
+    for (var j = 1; j <= 13; j += 1) {
+      var cardName = assignCardNames(j);
+      var cardNum = assignCardNumbers(j);
+      if (cardName == "ace") {
+        cardNum = 11;
+      }
+      //console.log(cardname);
+      var card = {
+        name: cardName,
+        suit: suits[i],
+        number: cardNum,
+        rank: j,
+      };
+      //console.log(card);
+      deck.push(card);
+    }
+  }
+  return deck;
+};
+
+// Assign Card Names (ace, jack, queen,king)
+// insert rank into input
+var assignCardNames = function (input) {
+  if (input == 1) {
+    return "ace";
+  } else if (input == 11) {
+    return "jack";
+  } else if (input == 12) {
+    return "queen";
+  } else if (input == 13) {
+    return "king";
+  } else {
+    return String(input);
+  }
+};
+
+// Assign Card Numbers (1,11,12,13 -> Number 1 and 10)
+var assignCardNumbers = function (input) {
+  if (input == 1) {
+    return 1;
+  } else if (input == 11) {
+    return 10;
+  } else if (input == 12) {
+    return 10;
+  } else if (input == 13) {
+    return 10;
+  } else {
+    return input;
+  }
+};
+
+// Shuffle a deck - Inputs: Deck (array type)
+var shuffleDeck = function (input) {
+  var i = 0;
+  while (i < input.length) {
+    var randomNumber = randomNum();
+    var proxycard_i = deck[i];
+    input[i] = input[randomNumber];
+    input[randomNumber] = proxycard_i;
+    i += 1;
+  }
+  return input;
+};
+
+// Get a random number
+var randomNum = function () {
+  return Math.floor(Math.random() * 52);
+};
+
+// GAME BEGINS HERE
+
+var gamestate = "START_STATE"; // START_STATE, HIT_STATE, STAND_STATE
+var playerHand = [];
+var computerHand = [];
+var deck = makeDeck();
+var shuffledDeck = shuffleDeck(deck);
+var playerHandValue = 0;
+var computerHandValue = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //check gamestate
+  if (checkCorrectInput(input)) {
+    return `Please enter Hit or Stand`;
+  } else {
+    checkGameState(input);
+  }
+  console.log("Check initial game state:" + gamestate);
+
+  if (gamestate == "START_STATE") {
+    gamestate = "PENDING_PLAYER_DECISION";
+    return startInitialGame();
+  }
+
+  if (gamestate == "HIT_STATE") {
+    var output = init_hit_state();
+    // console.log("Player hits, 1 card drawn, check deck: ");
+    // console.log(shuffledDeck);
+    return output;
+  } else if (gamestate == "STAND_STATE") {
+    console.log("Reach stand state");
+    init_stand_state();
+
+    var outcome = compareHandValues(playerHand, computerHand);
+    var myOutPut = GameOutcome(outcome);
+    resetgame();
+    return myOutPut;
+  }
+};
+
+/// 2. GAME STATE FUNCTION
+// Initial game state
+// Draw 2 cards for player and computer
+// Updates the drawn deck
+// Check for blackjack conditions
+var startInitialGame = function () {
+  for (var i = 0; i < 2; i += 1) {
```

Comment:
You can refactor this part into a separate function and maybe that receives the number of players
```js
function drawCardsForPlayers(numberOfPlayers) {
// logic here
}
```